src/ansys/geometry/core/geometry/curves/trimmed_curve.py

```diff
@@ -0,0 +1,209 @@
+
+        A parameter of 0 would correspond to the start of the curve, while a parameter of 1 would
+        correspond to the end of the curve.
+
+        Parameters
+        ----------
+        param : Real
+            Parameter in the proportional range [0,1].
+
+        Returns
+        -------
+        CurveEvaluation
+            The resulting curve evaluation.
+        """
+        bounds = self.interval
+        return self.geometry.evaluate(bounds.start + bounds.get_span() * param)
+
+    def intersect_curve(self, other: "TrimmedCurve") -> List[Point3D]:
+        """
+        Intersect this trimmed curve with another to receive the points of intersection.
+
+        If the curves do not intersect, an empty list is returned.
+
+        Parameters
+        ----------
+        other : TrimmedCurve
+            The trimmed curve to intersect with.
+
+        Returns
+        -------
+        List[Point3D]
+            All points of intersection between the curves.
+        """
+        if self._grpc_client is None:
+            raise ValueError(
+                """This TrimmedCurve was not initialized with a grpc client,
+                so this method cannot be called."""
+            )
+        first = trimmed_curve_to_grpc_trimmed_curve(self)
+        second = trimmed_curve_to_grpc_trimmed_curve(other)
+        res = self._commands_stub.IntersectCurves(
+            IntersectCurvesRequest(first=first, second=second)
+        )
+        if res.intersect is False:
+            return []
+        return [Point3D([point.x, point.y, point.z]) for point in res.points]
+
+    def __repr__(self) -> str:
+        """Represent the trimmed curve as a string."""
+        return (
+            f"TrimmedCurve(geometry: {type(self.geometry)}, start: {self.start}, end: {self.end}, "
+            f"interval: {self.interval}, length: {self.length})"
+        )
+
+
+class ReversedTrimmedCurve(TrimmedCurve):
+    """
+    Represents a reversed TrimmedCurve.
+
+    When a curve is reversed, its start and end points are swapped, and parameters for evaluations
+    are handled to provide expected results conforming to the direction of the curve. For example,
+    evaluating a TrimmedCurve proportionally at 0 would evaluate at the start point of the curve,
+    but evaluating a ReversedTrimmedCurve proportionally at 0 will evaluate at what was previously
+    the end point of the curve but is now the start point.
+
+    Parameters
+    ----------
+    geometry : Curve
+        The underlying mathematical representation of the curve.
+    start : Point3D
+        The original start point of the curve.
+    end : Point3D
+        The original end point of the curve.
+    interval : Interval
+        The parametric interval that bounds the curve.
+    length : Quantity
+        The length of the curve.
+    grpc_client : GrpcClient, default: None
+        Optional gRPC client that is required for advanced functions such as `intersect_curve()`.
```

Comment:
```suggestion
        Underlying mathematical representation of the curve.
    start : Point3D
        Original start point of the curve.
    end : Point3D
        Original end point of the curve.
    interval : Interval
        Parametric interval that bounds the curve.
    length : Quantity
        Length of the curve.
    grpc_client : GrpcClient, default: None
        gRPC client that is required for advanced functions such as `intersect_curve()`.
```
As earlier, use "functions" or "methods"?